### PR TITLE
MQE: only reduce resolution of delta in `rate` and `increase` once

### PR DESCRIPTION
**Pull request by charleskorn**
#### What this PR does

This PR modifies the behaviour of `rate` and `increase` to only reduce the resolution of the computed delta once, rather than every time the schema reduces in the input range.

This has no noticeable performance impact in our benchmarks as we don't exercise this case, as it rarely happens in practice.

#### Which issue(s) this PR fixes or relates to

Follow up to https://github.com/grafana/mimir/pull/9518#discussion_r1787164808

#### Checklist

- [n/a] Tests updated.
- [n/a] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [n/a] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: pkg/streamingpromql/functions/rate_increase.go [hunk 1]
```diff
@@ -114,8 +114,8 @@ func histogramRate(isRate bool, step types.RangeVectorStepData, hHead []promql.H
 				return histogram.ErrHistogramsIncompatibleSchema
 			}
 
-			if p.H.Schema < delta.Schema {
-				delta = delta.CopyToSchema(p.H.Schema)
+			if p.H.Schema < desiredSchema {
+				desiredSchema = p.H.Schema
```
Comment (jhesketh) [inline pkg/streamingpromql/functions/rate_increase.go:118]: do we need to copy delta to the desired schema at each counter reset (ie before adding the previous value)?

**Review by jhesketh — COMMENTED** [review]
Thanks for following this up!

File: pkg/streamingpromql/functions/rate_increase.go (hunk 1 above)
Comment (charleskorn) [reply pkg/streamingpromql/functions/rate_increase.go:118]: No - `Add` handles the case where the histograms have different schemas and will return a histogram with the lower of the two schemas (ie. the lower resolution).

Comment (jhesketh) [reply pkg/streamingpromql/functions/rate_increase.go:118]: Thanks, forgot about that.

**Review by jhesketh — APPROVED** [review]

**Pull request merged**